osfoffline/sync/ext/auditor.py

```diff
@@ -273,26 +318,38 @@ def _collect_node_local(self, root, acc, db_map):
                 self._collect_node_local(child, acc, db_map)
             else:
                 rel_path = str(child).replace(self.user_folder, '')
-                acc[rel_path] = Audit(
-                    db_map.get(rel_path, NULL_AUDIT).fid,
+                db_entry = db_map.get(rel_path, NULL_AUDIT)
+                # Local file may be aliased. If there is a matching DB object, use that to key local file audits
+                #   on the original name, to facilitate comparison with remote files.
+                path_key = db_entry.fobj.rel_path_unaliased if db_entry.fobj is not None else rel_path
+                acc[path_key] = Audit(
+                    db_entry.fid,
                     hash_file(child),
-                    rel_path
+                    path_key
                 )
         return acc
 
     def _diff(self, source, target):
         # source == snapshot
         # target == ref
-        id_target = {v.fid: k for k, v in target.items()}
-        id_source = {v.fid: k for k, v in source.items()}
 
-        created = set(source.keys()) - set(target.keys())
-        deleted = set(target.keys()) - set(source.keys())
+        # Filter out any alias entries, as they are duplicates of something tracked under another name
+        id_source = {v.fid: k for k, v in source.items() if not v.is_alias}
+        id_target = {v.fid: k for k, v in target.items() if not v.is_alias}
+
+        source_keys = set(k for k, v in source.items() if not v.is_alias)
+        target_keys = set(k for k, v in target.items() if not v.is_alias)
+
+        keys_in_both = source_keys & target_keys
+
+        created = source_keys - target_keys
+        deleted = target_keys - source_keys
 
-        for i in set(source.keys()) & set(target.keys()):
-            if source[i].fid != target[i].fid:
-                created.add(i)
-                deleted.add(i)
+        for k in keys_in_both:
+            if source[k].fid != target[k].fid and target[k].is_alias:
```

Comment:
For posterity: We really need a clearer understanding of the consequences of this. My understanding is that `keys_in_both` will contain no elements which are aliased in source or target. Consequently this branch would never be run.